File: 30_days_C#_HackerRank_Challenge/Class_VS_Instance.cs
using System;
using System.Collections.Generic;
using System.IO;

class Person {
    public int age;     
	public Person(int initialAge) 
    {
        if(initialAge < 0)
        {
            age = 0;
            Console.WriteLine("Age is not valid, setting age to 0.");
        }
        else
        {
         age = initialAge;   
        }
     }
     public void amIOld() 
     {
        if(age < 13)
        {
            Console.WriteLine("You are young.");
        }
        else if(age >= 13 && age <18)
        {
            Console.WriteLine("You are a teenager.");
        }
        else
        {
            Console.WriteLine("You are old.");
        }
     }

     public void yearPasses() {
        age++;
     }

static void Main(String[] args) {
        int T=int.Parse(Console.In.ReadLine());
        for (int i = 0; i < T; i++) {
            int age=int.Parse(Console.In.ReadLine());
            Person p=new Person(age);
             p.amIOld();
                for (int j = 0; j < 3; j++) {
                  p.yearPasses();
                }
                p.amIOld();
                Console.WriteLine();
        }
    }
}

File: 30_days_C(modern)_HackerRank_Challenge/Class_VS_Instance.c
#include <stdio.h>

typedef struct {
    int age;
} Person;

void initPerson(Person *p, int initialAge) {
    if (initialAge < 0) {
        p->age = 0;
        printf("Age is not valid, setting age to 0.\n");
    } else {
        p->age = initialAge;
    }
}

void amIOld(Person *p) {
    if (p->age < 13)
        printf("You are young.\n");
    else if (p->age < 18)
        printf("You are a teenager.\n");
    else
        printf("You are old.\n");
}

void yearPasses(Person *p) {
    p->age++;
}

int main() {
    int T, age;
    scanf("%d", &T);
    for (int i = 0; i < T; i++) {
        scanf("%d", &age);
        Person p;
        initPerson(&p, age);
        amIOld(&p);
        for (int j = 0; j < 3; j++)
            yearPasses(&p);
        amIOld(&p);
        printf("\n");
    }
    return 0;
}
